### PR TITLE
get_demands_endpoint

### PR DESCRIPTION
**Pull request by jiivan**
fixes: #404

### CODE REVIEW

File: core/market/decentralized/src/db/dao/demand.rs [hunk 1]
```diff
@@ -43,19 +44,29 @@ impl<'c> DemandDao<'c> {
...
-                .filter(dsl::insertion_ts.lt(insertion_ts))
+            let mut query = dsl::market_demand
                 .filter(dsl::expiration_ts.ge(validation_ts))
                 .order_by(dsl::creation_ts.asc())
-                .load::<Demand>(conn)?)
+                .into_boxed();
+
+            if let Some(ts) = insertion_ts {
+                // we querying less then here and less equal in Offers
+                // not to duplicate pair subscribed at the very same moment
```
Comment (nieznanysprawiciel) [inline core/market/decentralized/src/db/dao/demand.rs:61]: Outdated log??
`Offers`

File: core/market/decentralized/src/matcher/error.rs [hunk 2]
```diff
@@ -26,6 +26,14 @@ pub enum QueryOffersError {
     IdentityError(#[from] IdentityError),
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum QueryDemandsError {
+    #[error("Failed to get Demands. Error: {0}.")]
+    DbError(#[from] DbError),
+    #[error("Failed to list Demands based on identity. Error: {0}.")]
+    IdentityError(#[from] IdentityError),
```
Comment (nieznanysprawiciel) [inline core/market/decentralized/src/matcher/error.rs:34]: How can you get IdentityError when quering demands?

**Review by nieznanysprawiciel — DISMISSED** [review]

File: core/market/decentralized/src/db/dao/demand.rs (hunk 1 above)
Comment (jiivan) [reply core/market/decentralized/src/db/dao/demand.rs:61]: Altough I just moved this line :) it still looks relevant:
https://github.com/golemfactory/yagna/blob/3f8f7f9d7472e2bbe44076636eab58a2c1889958/core/market/decentralized/src/db/dao/offer.rs#L88-L90

Comment (nieznanysprawiciel) [reply core/market/decentralized/src/db/dao/demand.rs:61]: Ok, but there's `Offer` and we are dealing with `Demands`

File: core/market/decentralized/src/matcher/error.rs (hunk 2 above)
Comment (jiivan) [reply core/market/decentralized/src/matcher/error.rs:34]: I just copied it from offer. It's not needed - removed.

**Review by nieznanysprawiciel — APPROVED** [review]

**Pull request merged**